packages/core/test/result-summary.test.ts

```diff
@@ -48,3 +55,126 @@ describe('ServerInfo', () => {
     }
   )
 })
+
+describe('Notification', () => {
+  describe('.severityLevel', () => {
+    it.each(getValidSeverityLevels())('should fill severityLevel with the rawSeverityLevel equals to %s', rawSeverityLevel => {
+      const rawNotification = {
+        severity: rawSeverityLevel
+      }
+
+      const notification = new Notification(rawNotification)
+
+      expect(notification.severityLevel).toBe(rawSeverityLevel)
+      expect(notification.rawSeverityLevel).toBe(rawSeverityLevel)
+    })
+
+    it.each([
+      'UNKNOWN',
+      null,
+      undefined,
+      'I_AM_NOT_OKAY',
+      'information'
+    ])('should fill severityLevel UNKNOWN if the rawSeverityLevel equals to %s', rawSeverityLevel => {
+      const rawNotification = {
+        severity: rawSeverityLevel
+      }
+
+      const notification = new Notification(rawNotification)
+
+      expect(notification.severityLevel).toBe('UNKNOWN')
+      expect(notification.rawSeverityLevel).toBe(rawSeverityLevel)
+    })
+  })
+
+  describe('.category', () => {
+    it.each(getValidCategories())('should fill category with the rawCategory equals to %s', rawCategory => {
+      const rawNotification = {
+        category: rawCategory
+      }
+
+      const notification = new Notification(rawNotification)
+
+      expect(notification.category).toBe(rawCategory)
+      expect(notification.rawCategory).toBe(rawCategory)
+    })
+
+    it.each([
+      'UNKNOWN',
+      undefined,
+      null,
+      'DUNNO',
+      'deprecation'
+    ])('should fill category with UNKNOWN the rawCategory equals to %s', rawCategory => {
+      const rawNotification = {
+        category: rawCategory
+      }
+
+      const notification = new Notification(rawNotification)
+
+      expect(notification.category).toBe('UNKNOWN')
+      expect(notification.rawCategory).toBe(rawCategory)
+    })
+  })
+})
+
+describe('notificationSeverityLevel', () => {
+  it('should have keys equals to values', () => {
+    for (const [key, value] of Object.entries(notificationSeverityLevel)) {
+      expect(key).toEqual(value)
+    }
+  })
+
+  it('should values be assignable to NotificationSeverityLevel', () => {
```

Comment:
```suggestion
  it('should have values assignable to NotificationSeverityLevel', () => {
```
(more similar further down)